README.md

```diff
@@ -232,33 +216,135 @@ For example:
+You need only tick the `public_repo` permission when generating this.
+
+Alternatively, you can configure an Octokit instance without authentication using the grade `fluid.pipelines.withUnauthOctokit`.
+This is suitable for lightweight local testing but note that GitHub will limit the use of such requests to 60 per hour
+as described at [Resources in the REST API](https://docs.github.com/en/rest/overview/resources-in-the-rest-api).
+
+Either of `fluid.pipelines.withOctokit` or `fluid.pipelines.withUnauthOctokit` will construct a `{fluid.octokit}`
+instance in the pipeline that can be resolved by pipeline elements such as `fluid.fetchGitCSV`,
+`fluid.dataPipe.commitMultipleFiles`, etc.
+
+#### Running a pipeline using a run configuration
+
+The library supports a simple JSON structure which allows for the encoding of a particular pipeline run - that is,
+which pipelines should be loaded and merged together. This can then be executed using the library's CLI driver named
+`runFluidPipeline` by giving it this file as an argument. A run configuration has the following structure:
+
+````text
+{
+    loadDirectory: {String|String[]} [optional] - One or an array of module-qualified names of directories in which
+                                     all files are to be loaded as pipelines
+    loadPipeline: {String|String[]} [optional] One or an array of module-qualified names of individual pipeline
+    execPipeline: {String} A single pipeline grade names, loaded by the above directives, which will be executed
+    execMergedPipeline: {String[]} Multiple pipeline grade names which will be merged together and executed
+````
+
+For example, the embedded demo driver at [demoDriver](../demo/runDemo.json5) looks as follows:
+
+````text
+{ // Run configuration file for demonstration pipeline
+  // Can be run directly from command line via runFluidPipeline %forgiving-data/runDemo.json5
+    loadDirectory: "%forgiving-data/demo/pipelines",
+    execMergedPipeline: ["fluid.pipelines.WeCount-ODC-synthetic", "fluid.pipelines.WeCount-ODC-fileOutput"]
+}
+````
+
+Here are some possible ways of running this run configuration:
+
+##### As an npm script of a module for which `forgiving-data` is a dependency
+
+````text
+    "scripts": {
+        "runFluidPipeline": "runFluidPipeline %forgiving-data/demo/runDemo.json5",
+    }
+````
+
+From the hosting module you can then run
+
+    npm run runFluidPipeline
+
+##### If forgiving-data is installed as a global module using npm install -g forgiving-data
+
+    runFluidPipeline %forgiving-data/demo/runDemo.json5
+
+##### Via a local npm script as provided in this module
+
+````text
+    "scripts": {
+        "runFluidPipeline": "node src/cli.js"
     }
 ````
 
+From this module you can then run
+
+    npm run runFluidPipeline -- %forgiving-data/demo/runDemo.json5
```

Comment:
Characters `--` in this command doesn't seem necessary.

Reply:
But they are essential - https://stackoverflow.com/questions/11580961/sending-command-line-arguments-to-npm-script

Reply:
A nice discussion. I was wondering why the command works without `--` on my computer. The discussion clarified - `If your param does not start with - or --, then having an explicit -- separator is not needed; but it's better to do it anyway for clarity.`

Ya, it's better to keep `--`.